### PR TITLE
trackForCleanup all devices, and await device.lost while cleaning up

### PR DESCRIPTION
**Pull request by kainino0x**
Dry run (prior to rebase) to check for breakage: <https://dawn-review.googlesource.com/c/dawn/+/196138?tab=checks>

Issue: #1752

<hr>

**Requirements for PR author:**

- [n/a] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [n/a] Tests are properly located in the test tree.
- [n/a] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [n/a] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/web_platform/reftests/canvas_display_after_device_lost.html.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ void (async () => {
 
   const adapter = await navigator.gpu.requestAdapter();
   assert(adapter !== null);
+  // eslint-disable-next-line no-restricted-syntax
```
Comment (greggman) [inline src/webgpu/web_platform/reftests/canvas_display_after_device_lost.html.ts:13]: Should we just add a .eslintrc.js to the reftests folder that turns off this rule?

**Review by greggman — APPROVED** [review]
This is fine, just a few questions 1 here and in-line

You were okay before with making all the code call `t.createXXXTracked` function to create buffers, textures, querySets. What about `t.requestDeviceTracked(adapter, deviceDesc)`?

File: src/stress/adapter/device_allocation.spec.ts [hunk 2]
```diff
@@ -290,6 +290,8 @@ implicitly keep the device in scope.`
     const kNumDevices = 10_000;
     for (let i = 1; i <= kNumDevices; ++i) {
       await (async () => {
+        // No trackForCleanup because it would prevent the GPUDevice from being GCed.
```
[the remark this answers is not in the document]
Comment (greggman) [reply src/stress/adapter/device_allocation.spec.ts:293]: Never mind about `WeakSet`. I forgot it doesn't have a `values` function.

File: src/webgpu/web_platform/reftests/canvas_display_after_device_lost.html.ts (hunk 1 above)
Comment (kainino0x) [reply src/webgpu/web_platform/reftests/canvas_display_after_device_lost.html.ts:13]: Oh yes, that would be a good idea. I sort of considered it but forgot (I'd have to turn off the entire no-restricted-syntax rule but that seems fine.)

**Comment by kainino0x** [comment]
> You were okay before with making all the code call `t.createXXXTracked` function to create buffers, textures, querySets. What about `t.requestDeviceTracked(adapter, deviceDesc)`?

A lot of the tests that create their own device don't inherit from GPUTestBase. Not sure exactly how many but seemed like it was enough that it wouldn't be worth having a helper for only those.

**Comment by kainino0x** [comment]
Though I guess nothing prevents me from putting it in Fixture.

**Comment by kainino0x** [comment]
https://github.com/gpuweb/cts/commit/a5fad1b013381c459e576738e6eb1785d38e4e4d Slightly shorter but I'm not sure it's really worth it?

(I was going to open it as a separate PR but GitHub seems to be having an outage right now that prevents me from landing this PR.)

File: src/stress/adapter/device_allocation.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kainino0x) [reply src/stress/adapter/device_allocation.spec.ts:293]: Had the same thought and realization :)

**Pull request merged**